lib/diagnostic.js

```diff
@@ -5,17 +5,29 @@
 // the minimum value in an array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let minimum = array.reduce(min);`
-const min = function (/* previous, current */) {
-
+const min = function ( previous, current )
+{
+  let result = array[0];
+  let minimum = function ( array, callback )
+  {
+    return array.forEach( e, i, a )
+    {
+      if( result[i] > array[i] )
+      {
+        resut[i] = array[i];
+      }
+    }
+  };
 };
 
 // Question 2
 // Write a callback for reduce that produces
 // the sum of the values in the array
 // this callback will be passed by the tests to the reduce method, e.g.
 // `let total = array.reduce(sum);`
-const sum = function (/* previous, current */) {
-
+const sum = function ( previous, current )
+{
+  let total = array.reduce( (previous, current ) => previous, current, 0;
 };
```

Comment:
`reduce` is a method available in JavaScript that takes in a callback. The question asks you to write the callback that would go _inside_ the `reduce` method.